python/main.py

```diff
@@ -19,16 +52,75 @@
 )
 
 
+
+
 @app.get("/")
 def root():
     return {"message": "Hello, world!"}
 
+#STEP3-3, 4-1
+@app.get("/items")
+def get_items_from_database():
+
+    conn = sqlite3.connect(DATABASE)
+    cursor = conn.cursor()
+
+    cursor.execute("SELECT * FROM items")
+    items = cursor.fetchall()
+
+    cursor.close()
+    conn.close()
+
+    items_list = [{"id": row[0], "name": row[1], "category": row[2], "image_name": row[3]} for row in items]
 
+    print("Items list", items_list)
+
+    return {"items": items_list}
+
+    
+#STEP3-2, 3-4, 4-1, 4-3
 @app.post("/items")
-def add_item(name: str = Form(...)):
-    logger.info(f"Receive item: {name}")
-    return {"message": f"item received: {name}"}
+def add_item(name: str = Form(...), category_name: str = Form(...), image: UploadFile = File(...)):
+    #STEP4-1
+
+    image_filename = get_image_filename(image)
+
+    conn = sqlite3.connect(DATABASE)
+    cursor = conn.cursor()
+
+    cursor.execute ("SELECT id FROM categories WHERE name = ?", (category_name,)) 
+    category_row = cursor.fetchone()
 
+    if category_row == None:
```

Comment:
that's good! as an alternative you can look at this: 
https://www.sqlite.org/lang_upsert.html